### DISCUSSION TITLE
How to implement the function of vim jump list?

### DISCUSSION THREAD

**Discussion by longguzzz** [gdh1995/vimium-c#1000] [Q&A]
Is there any way to do the same thing as <c-o> and <c-i> do in vim?

**Comment by gdh1995** [comment]
No same commands but there's a similar feature in Vimium C.

 You may try `Marks.activateCreate` and `Marks.activate`, to create marks and then jump back to them.

> **Using marks** (https://github.com/philc/vimium):
> ```
> ma, mA  set local mark "a" (global mark "A")
> `a, `A  jump to local mark "a" (global mark "A")
> ``      jump back to the position before the previous jump
>           -- that is, before the previous gg, G, n, N, / or `a
> ```
>
> **Swapping global and local marks** (https://github.com/philc/vimium/wiki/Tips-and-Tricks#swapping-global-and-local-marks)
> In Vim, upper-case letters are global marks, whereas lower-case letters are local marks.
> 
> In a browser, global marks are arguably more useful than local marks. Therefore, Vimium allows you to swap these two:
> ```
> map X Marks.activateCreateMode swap
> map Y Marks.activateGotoMode swap
> ```

And, Vimium C has a new feature in local mark commands:
- local marks support "count prefix", so ``` 2`` ``` will create or jump to a special mark named `2`, which is different from the one created by ``` `` ```.
